text/0066-simulation-time.md

```diff
@@ -0,0 +1,119 @@
+- Start Date: (fill in with date at which the RFC is merged, YYYY-MM-DD)
+- RFC PR: [amaranth-lang/rfcs#66](https://github.com/amaranth-lang/rfcs/pull/66)
+- Amaranth Issue: [amaranth-lang/amaranth#0000](https://github.com/amaranth-lang/amaranth/issues/0000)
+
+# Simulation time
+
+## Summary
+[summary]: #summary
+
+Add a type for representing time durations and simulator methods to query the elapsed time.
+
+## Motivation
+[motivation]: #motivation
+
+In [RFC #36](0036-async-testbench-functions.md), there were a plan to introduce a `.time()` method to query the current simulation time, but it got deferred due to the lack of a suitable return type.
+
+Internally simulation time is an integer number of femtoseconds, but exposing that directly is not ergonomic and making it a `float` of seconds sacrifices precision the longer a simulation runs.
+
+Also, to quote @whitequark: Time is not a number.
+
+## Guide- and reference-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+A new type `amaranth.sim.Period` is introduced.
+It is immutable, has no public constructor and is instead constructed through the following classmethods:
+- `.s(duration: numbers.Real)`
+- `.ms(duration: numbers.Real)`
+- `.us(duration: numbers.Real)`
+- `.ns(duration: numbers.Real)`
+- `.ps(duration: numbers.Real)`
+- `.fs(duration: numbers.Real)`
+- `.Hz(frequency: numbers.Real)`
+- `.kHz(frequency: numbers.Real)`
+- `.MHz(frequency: numbers.Real)`
+- `.GHz(frequency: numbers.Real)`
+  - The argument will be scaled according to the SI prefix on the method and the `duration` or reciprocal of `frequency` then used to calculate the closest integer femtosecond representation.
+
+To convert it back to a number, the following properties are available:
```

Comment:
Can we provide a convenience wrappers over time getters, to request the period rounded to int? If we specify time, then usually we know which unit is needed to get the exact int value. In such cases there will be always needed explicit cast which will produce boilerplate.

So I would like to propose to add:
- `.iseconds` -> int
- `.imiliseconds` -> int
- and so on

(of course the above function names are some example, which should probably by changed for something better)

Reply:
I don't see that as valuable due to how rare reading simulation time is expected to be.